### ISSUE TITLE
Nuxt 3 only

### ISSUE THREAD

**Issue by atinux** [nuxt-modules/color-mode#173]
I want to take the opportunity to simplify the code base and make it nuxt 3 only with some breaking changes:

- `classSuffix` will be `''` by default

That's it for now 😃

**Comment by P4sca1** [comment]
@Atinux For your planned refactoring: Could you consider bringing back cookie storage, so that we know the user preference during SSR? This is useful to set the correct `theme-color` header, or to render the correct icon for the color mode menu on the server for example. Otherwise it leads to a flash.
Maybe there could be an option to decide which storage to use.

Also, there is an issue for a while related to the typings, which destroys Nuxt DX. I hope it can be fixed by only using Nuxt 3 syntax. https://github.com/nuxt-modules/color-mode/issues/138

**Comment by Atinux** [comment]
Thanks for linking as well @P4sca1 

We actually add a script in the `<head>` to update the value of the `data-theme` and `class`. This way you should not see any styling issue.

We had some issues with Cookies and Safari on mobile too: https://github.com/nuxt-modules/color-mode/issues/38

When do you have the flash?

**Comment by P4sca1** [comment]
E.g. you have theme-color blue for dark mode and white for light mode.
When you reload the page with dark mode enabled, the header in safari (compact mode) will first be white and  then switch to blue after the meta value has been updated on the client.
That is, because the server can't know the user preference during SSR and does not know which value to set for the theme-color meta tag.

**Comment by P4sca1** [comment]
Yes, I am using a Node.js server to deploy my app. I agree that for some deployments, cookies are not an option. So maybe the storage backend could be configurable in some way.

**Comment by cdwmhcc** [comment]
Any updates?

**Comment by Atinux** [comment]
@cdwmhcc as soon as I get some time for it.

**Comment by dargmuesli** [comment]
v4 should have this issue resolved :) cc @danielroe
